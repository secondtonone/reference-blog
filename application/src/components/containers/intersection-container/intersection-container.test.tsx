import 'jsdom-global/register';
import renderer from 'react-test-renderer';
import IntersectionContainer from '.';

const intersectionObserverMock = () => ({
  observe: () => null
});

global.IntersectionObserver = jest.fn().mockImplementation(intersectionObserverMock);

describe('Match snapshots IntersectionContainer', () => {
  it('default', () => {
    const tree = renderer.create(
      <IntersectionContainer onIntersection={() => ({})}>
        IntersectionContainer
      </IntersectionContainer>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('intersected once', () => {
    const tree = renderer.create(
      <IntersectionContainer once onIntersection={() => ({})}>
        IntersectionContainer
      </IntersectionContainer>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
