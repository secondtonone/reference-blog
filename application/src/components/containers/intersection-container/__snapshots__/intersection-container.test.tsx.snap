// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match snapshots IntersectionContainer default 1`] = `
<div
  className="box-adaptive__BoxStyled-sc-g2wy39-0"
  data-test="intersection-container"
>
  IntersectionContainer
</div>
`;

exports[`Match snapshots IntersectionContainer intersected once 1`] = `
<div
  className="box-adaptive__BoxStyled-sc-g2wy39-0"
  data-test="intersection-container"
>
  IntersectionContainer
</div>
`;
